LocalPageStore delete pageId could ignore parent path NoSuchFileException when parent path is deleted by other thread in a benign race

**Alluxio Version:**
2.7.3

**Describe the bug**
Presto uses the alluxio local ssd cache, and the log sees a lot of alluxio delete page failures because the page's parent directory does not exist, NoSuchFileException.
Exception stack code analysis, when writing to the cache, because the storage reaches the set capacity limit, the old page must be deleted first.
After deleting the old page file, an attempt is made to delete the page's parent directory.
In a concurrency scenario, the parent directory of the page may just be deleted by another thread that deletes the same FileId and has a different PageIndex. At this point, the operation on the parent directory is reported to NoSuchFileException.
Delete the old page encountered IOException will return PutResult.OTHER, put cache directly failed and will not be retried.



**To Reproduce**
persto log:
2022-09-21T03:58:04.256+0800    ERROR   20220920_195803_66351_y2msc.2.0.6-4-470105      alluxio.client.file.cache.LocalCacheManager     Failed to delete page PageId{FileId=7e70de79febe84dbbd94a103237977fc, PageIndex=5} from pageStore
java.io.IOException: java.util.concurrent.ExecutionException: java.nio.file.NoSuchFileException: /home/presto/cache/data/LOCAL/1048576/173/9cd49f00622352111b7216e7d35b98ca
        at alluxio.client.file.cache.TimeBoundPageStore.delete(TimeBoundPageStore.java:126)
        at alluxio.client.file.cache.LocalCacheManager.putAttempt(LocalCacheManager.java:411)
        at alluxio.client.file.cache.LocalCacheManager.putInternal(LocalCacheManager.java:295)
        at alluxio.client.file.cache.LocalCacheManager.put(LocalCacheManager.java:256)
        at alluxio.client.file.cache.CacheManagerWithShadowCache.put(CacheManagerWithShadowCache.java:113)
        at alluxio.client.file.cache.NoExceptionCacheManager.put(NoExceptionCacheManager.java:51)
        at alluxio.client.file.cache.LocalCacheFileInStream.readInternal(LocalCacheFileInStream.java:185)
        at alluxio.client.file.cache.LocalCacheFileInStream.positionedRead(LocalCacheFileInStream.java:228)
        at alluxio.hadoop.HdfsFileInputStream.read(HdfsFileInputStream.java:153)
        at alluxio.hadoop.HdfsFileInputStream.readFully(HdfsFileInputStream.java:170)
        at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:107)
        at com.facebook.presto.hive.orc.HdfsOrcDataSource.readInternal(HdfsOrcDataSource.java:64)
        ...
Caused by: java.util.concurrent.ExecutionException: java.nio.file.NoSuchFileException: /home/presto/cache/data/LOCAL/1048576/173/9cd49f00622352111b7216e7d35b98ca
        at alluxio.shaded.client.com.google.common.util.concurrent.SimpleTimeLimiter.wrapAndThrowExecutionExceptionOrError(SimpleTimeLimiter.java:280)
        at alluxio.shaded.client.com.google.common.util.concurrent.SimpleTimeLimiter.callWithTimeout(SimpleTimeLimiter.java:167)
        at alluxio.client.file.cache.TimeBoundPageStore.delete(TimeBoundPageStore.java:114)
        ... 41 more
Caused by: java.nio.file.NoSuchFileException: /home/presto/cache/data/LOCAL/1048576/173/9cd49f00622352111b7216e7d35b98ca
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(UnixFileSystemProvider.java:432)
        at java.base/java.nio.file.Files.newDirectoryStream(Files.java:472)
        at alluxio.client.file.cache.store.LocalPageStore.delete(LocalPageStore.java:148)
        at alluxio.client.file.cache.TimeBoundPageStore.lambda$delete$2(TimeBoundPageStore.java:110)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        ... 3 more

**Expected behavior**
Seeing this exception occur up to a few hundred times a minute during peak periods affects write caching.
When throwing this exception, the old page has been deleted in both mMetaStore and mPageStore, and the parent directory does not exist, which means that the old page has been successfully deleted.
Maybe we can optimize this case, catch the exception  and continue to put the page operation.

**Urgency**
Affects write caching

**Are you planning to fix it**
yes~

**Additional context**

--- a/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java
+++ b/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java
@@ -27,6 +27,7 @@ import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.nio.file.DirectoryStream;
 import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
@@ -141,6 +142,11 @@ public class LocalPageStore implements PageStore {
     try (DirectoryStream<Path> stream = Files.newDirectoryStream(parent)) {
       if (!stream.iterator().hasNext()) {
         Files.delete(parent);
+      }
+    } catch (NoSuchFileException e) {
+      //Parent path is deleted by other thread in a benign race, ignore exception and continue
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Parent path is deleted by other thread, ignore and continue.", e);
       }
     }
   }

